.azure-pipelines/simplified_PR_test/README.md

```diff
@@ -0,0 +1,106 @@
+
+
+## Design
+### Impcated Area
+To take advantage of such code structure, we introduce a new term called `impacted area`, which represents the scope of PR testing.
+The `impacted area` can be defined by specific features, so that we can narrow down the scope into folders.
+
+This term can be elaborated as follows:
+- If the changes are confined to a specific feature folder, we can narrow the scope of testing to only include files within that folder.
+As files in other feature folders remain unaffected and do not require testing.
+- If the changes affect the common components, we cannot narrow the testing scope and must run all test scripts to ensure comprehensive coverage, as they are commonly used by other features.
+
+We can determine the impcated area using command `git diff`.
+
+### Distribute scripts to PR checkers
+In our new PR test, we will have multiple PR checkers classified by topology type.
+To distribute all required scripts for each PR checker, which means,
+these scripts should not only within the scope that we changed, but also meet the requirement of topology.
+
+We can suggest two approaches to achieve this:
+- One approach is by using the `--topology` parameter supported by pytest.
+It compares against the topology marked with `pytest.mark.topology` in script,
+and if the mark matches, the script is deemed necessary.
+However, this method triggers pytest's collection process for each script,
+leading to unnecessary time consumption, which is not expected.
+
+- Another approach is to collect and analyze all scripts before execution.
+Each script includes the `pytest.mark.topology` marker to indicate the applicable topology it can run on.
+We will perform a global scan of all test scripts in the impacted area to identify this marker and extract its value,
+which represents the topology type compatible with the script.
+After determining the valid topology for each script, we can distribute the script to corresponding PR checkers.
+This method eliminates unnecessary processes by executing only the on-demand scripts, resulting in reduced running time.
+
+### Implement dynamic instances
+Since the scope of PR testing is dynamic and determined by the impacted area,
+the number of instances required also needs to be dynamic to ensure cost efficiency.
+To achieve this, we must accurately estimate the total execution time in advance,
+allowing us to allocate the appropriate number of instances.
+This estimation can be achieved by analyzing historical data,
+which provides insights into execution times for similar scenarios.
+
+We now have a Kusto table that logs details about the execution of test cases,
+including the running time, date, results, and more.
+To determine the preset running time for each test script,
+we will calculate the average running time of the latest five run times.
+If no relevant records are found in Kusto, a default value(1800s per script) will be used for the preset running time.
```

Comment:
`1800s per script`, 1800s is 30 minutes, how to determine this?

Reply:
This time is to keep pace with Elastictest.

Reply:
Ok, lets' use it first.